src/Facets/AllBridgeFacet.sol

```diff
@@ -0,0 +1,148 @@
+    /// @param recipient The address of the token receiver after bridging.
+    /// @param destinationChainId The destination chain id.
+    /// @param receiveToken The token to receive on the destination chain.
+    /// @param nonce A random nonce to associate with the tx.
+    /// @param messenger The messenger protocol enum
+    struct AllBridgeData {
+        uint256 fees;
+        bytes32 recipient;
+        uint8 destinationChainId;
+        bytes32 receiveToken;
+        uint256 nonce;
+        MessengerProtocol messenger;
+    }
+
+    /// @notice Initializes the AllBridge contract
+    /// @param _allBridge The address of the AllBridge contract
+    constructor(IAllBridge _allBridge) {
+        allBridge = _allBridge;
+    }
+
+    /// @notice Bridge tokens to another chain via AllBridge
+    /// @param _bridgeData The bridge data struct
+    function startBridgeTokensViaAllBridge(
+        ILiFi.BridgeData memory _bridgeData,
+        AllBridgeData calldata _allBridgeData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        validateBridgeData(_bridgeData)
+        doesNotContainSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+    {
+        LibAsset.depositAsset(
+            _bridgeData.sendingAssetId,
+            _bridgeData.minAmount
+        );
+        _startBridge(_bridgeData, _allBridgeData);
+    }
+
+    /// @notice Bridge tokens to another chain via AllBridge
+    /// @param _bridgeData The bridge data struct
+    /// @param _swapData The swap data struct
+    /// @param _allBridgeData The AllBridge data struct
+    function swapAndStartBridgeTokensViaAllBridge(
+        ILiFi.BridgeData memory _bridgeData,
+        LibSwap.SwapData[] calldata _swapData,
+        AllBridgeData calldata _allBridgeData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        containsSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+        validateBridgeData(_bridgeData)
+    {
+        _bridgeData.minAmount = _depositAndSwap(
+            _bridgeData.transactionId,
+            _bridgeData.minAmount,
+            _swapData,
+            payable(msg.sender)
+        );
+        _startBridge(_bridgeData, _allBridgeData);
+    }
+
+    /// @notice converts an address to bytes
+    /// @param a The address to convert
+    function _toBytes(address a) internal pure returns (bytes memory) {
+        return abi.encodePacked(a);
+    }
+
+    /// @notice Bridge tokens to another chain via AllBridge
+    /// @param _bridgeData The bridge data struct
+    /// @param _allBridgeData The allBridge data struct for AllBridge specicific data
+    function _startBridge(
+        ILiFi.BridgeData memory _bridgeData,
+        AllBridgeData calldata _allBridgeData
+    ) internal {
+        bool isNative = _bridgeData.sendingAssetId == LibAsset.NATIVE_ASSETID;
```

Comment:
doesnt really make a difference but just for convention...LibAsset has a function called "isNativeAsset(_bridgeData.sendingAssetId)" that we would normally use in such case

And since the variable "isNative" is only used once, I would suggest to simplify the code and put the "isNativeAsset" function directly in the if condition

Reply:
👍